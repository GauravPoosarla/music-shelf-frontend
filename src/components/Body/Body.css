.body {
  display: flex;
  justify-content: center;
  flex-direction: column;
  background-color: #f5f5f5;
  min-height: 800px;
}

.loading-screen {
  font-weight: 200;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 30px;
  padding-bottom: 150px;
}

@media screen and (max-width: 768px) {
  .body {
    padding: 10px;
  }
  .loading-screen {
    font-size: 18px;
    padding-bottom: 80px;
  }
}
  