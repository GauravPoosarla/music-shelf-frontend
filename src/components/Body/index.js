import React from 'react';
import './Body.css';
import { useState } from 'react';
import { useEffect } from 'react';
import Button from '../Button';
import Card from '../Card';
import makeRequest from '../../utils/makeRequest';
import { GET_SONGS_DATA } from '../../constants/apiEndPoint';

function Body() {
  const [songs, setSongs] = useState([]);
  const [isClicked, setIsClicked] = useState(false);

  useEffect(() => {
    makeRequest(GET_SONGS_DATA(), {
      headers: { authorization: 'Bearer QWlzaHdhcnlhIE4=' },
    }).then((data) => {
      setSongs(data.data);
    });
  }, []);

  return (
    <div className="body">
      {!isClicked ? (
        <div className="loading-screen">
          <p>::</p>
          <p>seems a bit empty in here ...</p>
          <Button setIsClicked={setIsClicked} />
        </div>
      ) : (
        <div className="body-content">
          <div className="songs">
            {songs.map((song) => (
              <Card song={song} />
            ))}
          </div>
        </div>
      )}
    </div>
  );
}

export default Body;
