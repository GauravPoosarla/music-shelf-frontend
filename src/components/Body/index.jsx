import React from 'react';
import './Body.css';
import { useState } from 'react';
import { useEffect } from 'react';
import { Button } from '../Button';
import { useNavigate } from 'react-router-dom';

function Body() {
  const [isClicked, setIsClicked] = useState(false);
  const navigate = useNavigate();

  useEffect(() => {
    if (isClicked) {
      navigate('/songs');
    }
  }, [isClicked]);

  return (
    <div className="body">
      <div className="loading-screen">
        <p>:((</p>
        <p>seems a bit empty in here ...</p>
        <br />
        <Button setIsClicked={setIsClicked} />
      </div>
    </div>
  );
}

export default Body;
