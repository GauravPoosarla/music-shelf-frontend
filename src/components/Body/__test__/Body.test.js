import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { useNavigate } from 'react-router-dom';
import Body from '..';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

describe('Body', () => {
  it('should render the text when mounted', () => {
    render(<Body />);
    expect(screen.queryByText('seems a bit empty in here ...')).toBeTruthy();
  });

  it('should navigate to /songs when the button is clicked', () => {
    render(<Body />);
    const button = screen.getByText('Sync');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith('/songs');
  });

  it('should render correctly', () => {
    const { container } = render(<Body />);
    expect(container).toMatchSnapshot();
  });
});
