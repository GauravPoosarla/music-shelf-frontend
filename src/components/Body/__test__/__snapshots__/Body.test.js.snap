// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render correctly 1`] = `
<div>
  <div
    class="body"
  >
    <div
      class="loading-screen"
    >
      <p>
        :((
      </p>
      <p>
        seems a bit empty in here ...
      </p>
      <br />
      <div
        class="click"
      >
        <button>
          Sync
        </button>
      </div>
    </div>
  </div>
</div>
`;
